ui/src/app/shared/components/env-input/env-input.tsx

```diff
@@ -36,7 +36,7 @@ class State {
     public newItem: Item;
 }
 
-class ArrayInput extends React.Component<Props, State> {
+class EnvInput extends React.Component<Props, State> {
     constructor(props: Readonly<Props>) {
         super(props);
         this.state = {newItem: {name: '', value: ''}, items: props.items};
```

Comment:
I missed that when ArrayInput was introduced. This component does not require state since state is stored in the Form. State should be removed and items from properties should be used to render the component.